kolena/_experimental/dataset/_dataset.py

```diff
@@ -95,7 +131,7 @@ def _infer_datatype(df: pd.DataFrame) -> Union[pd.DataFrame, str]:
 def _to_serialized_dataframe(df: pd.DataFrame, column: str) -> pd.DataFrame:
     result = _dataframe_object_serde(df, _serialize_dataobject)
     if column == COL_DATAPOINT:
-        result[DATA_TYPE_FIELD] = _infer_datatype(df)
+        _add_datatype(result)
```

Comment:
Current logic has no issue, but I'm wondering if we should add a test to ensure `_to_serialized_dataframe` doesn't apply this logic to, say, serializing result dataframe.